Mathlib/Topology/MetricSpace/HausdorffDimension.lean

```diff
@@ -494,6 +494,78 @@ lemma hausdorffMeasure_of_finrank_lt [MeasurableSpace E] [BorelSpace E] {d : ℝ
   rw [dimH_univ_eq_finrank]
   exact mod_cast hd
 
+/-- The Hausdorff dimension of the unit interval is 1. -/
+theorem dimH_Icc01 : dimH (Icc (0 : ℝ) 1) = 1 := by
+  have : (interior (Icc (0 : ℝ) 1)).Nonempty := by simp
+  calc
+    dimH (Icc (0 : ℝ) 1) = ↑(finrank ℝ ℝ) := by
+        simpa [Module.finrank_self, ENNReal.coe_one] using Real.dimH_of_nonempty_interior this
+    _ = 1 := by simp
+
+omit [FiniteDimensional ℝ E]
+
+/-- The Hausdorff dimension of a non-degenerate segment in a real normed space is 1. -/
+theorem dimH_segment {x y : E} (h : x ≠ y) :
+    dimH (segment ℝ x y) = 1 := by
+  have seg_eq : segment ℝ x y =
+    (fun t : ℝ ↦ (1 - t) • x + t • y) '' Icc (0 : ℝ) 1 := segment_eq_image ℝ x y
+  set f : ℝ → E := fun t ↦ (1 - t) • x + t • y with hf
+  set s := Icc (0 : ℝ) 1 with hs
+  observe hxy : y - x ≠ 0
+  observe hC : ‖y - x‖₊ ≠ 0
+  have hL : LipschitzWith ‖y - x‖₊ f := by
+    refine LipschitzWith.of_dist_le_mul fun a b ↦ ?_
+    calc
+      dist (f a) (f b) = ‖(1 - a) • x + a • y - ((1 - b) • x + b • y)‖ := by simp [hf, dist_eq_norm]
+      _ = ‖(a - b) • (y - x)‖ := by
+        rw [@add_sub_add_comm, ← sub_smul, sub_sub_sub_cancel_left, ← sub_smul]
+        have : (b - a) • x = -((a - b) • x) := by
+          simp [← neg_smul]
+        rw [this, ← neg_smul, add_comm]
+        have : (a - b) • y + -(a - b) • x = (a - b) • (y - x) := by
+          simp [smul_sub]
+          rw [sub_smul, sub_smul, sub_smul, ← sub_smul]
+          observe : - (a • x - b • x) = (b • x - a • x)
+          rw [← this]
+          simp only [sub_eq_add_neg]
+        expose_names; rw [this]
+      _ = |a - b| * ‖y - x‖ := by rw [norm_smul, Real.norm_eq_abs]
+      _ ≤ ‖y - x‖ * dist a b := by simp [dist_eq_norm, mul_comm]
+  have hA : AntilipschitzWith ((‖y - x‖₊)⁻¹) f := by
+    refine AntilipschitzWith.of_le_mul_dist fun a b ↦ ?_
+    calc
+      dist a b = |a - b| := by simp [dist_eq_norm]
+      _ = ↑‖y - x‖₊⁻¹ * ‖(a - b) • (y - x)‖ := by
+        simp only [norm_smul, NNReal.coe_inv]
+        rw [mul_comm]
+        aesop
+      _ = ↑‖y - x‖₊⁻¹ * dist (f a) (f b) := by
+        simp [hf, dist_eq_norm]
+        left
+        rw [@add_sub_add_comm, ← sub_smul, sub_sub_sub_cancel_left, ← sub_smul]
+        have : (b - a) • x = -((a - b) • x) := by
+          simp [← neg_smul]
+        rw [this, ← neg_smul, add_comm]
+        have : (a - b) • y + -(a - b) • x = (a - b) • (y - x) := by
+          simp [smul_sub]
+          rw [sub_smul, sub_smul, sub_smul, ← sub_smul]
+          observe : - (a • x - b • x) = (b • x - a • x)
+          rw [← this]
+          simp only [sub_eq_add_neg]
+        expose_names; rw [this]
+      _ ≤ _ := le_rfl
+  have L : dimH (segment ℝ x y) ≤ 1 := by calc
```

Comment:
I am pretty sure this PR can also be golfed a bit. Here is one example: you can replace your calc block by
```
  have L : dimH (segment ℝ x y) ≤ 1 := by
    rw [seg_eq]
    grw [hL.dimH_image_le _]
    rw [hs, ← dimH_Icc01]
```
and subsequently compress that down to
`have L : dimH (segment ℝ x y) ≤ 1 := by grw [seg_eq, hL.dimH_image_le _, hs, ← dimH_Icc01]`.

The `grw` tactic is new (see [this zulip thread](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Announcement.3A.20New.20.60grw.60.20tactic/with/528538038) for some examples; for instance, you can rewrite with inequalities).
you should be able to
- do `rw 

Here's one example how you could gol